src/cache/tlru-cache.ts

```diff
@@ -0,0 +1,77 @@
+import { LRUCache } from './lru-cache';
+
+/**
+ * Time-aware, least-recently-used cache (TLRU). Variant of LRU where entries have valid lifetime.
+ * @param {number} maxSize - Maximum cache size
+ * @param {number} ttl - Time in milliseconds after which cache entry will evict itself
+ * @param {number} evictionInterval - Frequency of cache entries eviction check
+ **/
+export class TLRUCache extends LRUCache {
+  private readonly cacheEntriesTTLRegistry = new Map<string, Date>();
+  constructor(readonly maxSize: number, readonly ttl: number) {
+    super(maxSize);
+  }
+
+  private getCacheEntryEvictionTime(): Date {
+    return new Date(Date.now() + this.ttl);
+  }
+
+  private clearCacheEntryEvictionTimeIfExists(key: string): void {
+    if (this.cacheEntriesTTLRegistry.has(key)) {
+      this.cacheEntriesTTLRegistry.delete(key);
+    }
+  }
+
+  private setCacheEntryEvictionTime(key: string): void {
+    this.cacheEntriesTTLRegistry.set(key, this.getCacheEntryEvictionTime());
+  }
+
+  private resetCacheEntryEvictionTime(key: string): void {
+    this.clearCacheEntryEvictionTimeIfExists(key);
+    this.setCacheEntryEvictionTime(key);
+  }
+
+  private evictExpiredCacheEntries() {
```

Comment:
I think we only need to do this for `entries()`, `keys()`, and `values()`, which we could override here, call this, then delegate to the parent method.

`get()` can be done just in time and I don't think `set()` needs to care at all.